src/assets/src/hooks/useWebSocket.ts

```diff
@@ -57,7 +56,9 @@ export const useWebSocket = <T>(url: string, onUpdate: (content: T) => void, onD
             console.error(e);
             const errorName = e?.error?.name as string | undefined; // On Safari, Event is sometimes emitted instead of ErrorEvent
             const errorText = errorName ? ` (${errorName})` : "";
-            setError(new Error(`An unexpected error occurred${errorText}. Trying to reconnect...`));
+            const errorBeginning = `An unexpected error occurred${errorText}. `;
+            const errorEnding = ws.retryCount < numRetries ? 'Trying to reconnect...' : 'Unable to reconnect; please refresh the page.';
```

Comment:
I'm not exactly sure when `ws.retryCount` is incremented, so this may need some tweaking to ensure all three attempts have been made before the message changes.

Reply:
Yes, it's hard to validate the number of retries as there isn't an "onretry" event to attach to. I get three errors from `reconnecting-websocket-mjs.js:518` (WebSocket connection failed: ERR_CONNECTION_REFUSED), which seems to indicate three failed retries. Let's trust the library; the exact number of retries isn't important to the functionality aiming for anyway.